src/sinks/elasticsearch.rs

```diff
@@ -441,6 +441,7 @@ impl ElasticSearchCommon {
 
     fn signed_request(&self, method: &str, uri: &Uri, use_params: bool) -> SignedRequest {
         let mut request = SignedRequest::new(method, "es", &self.region, uri.path());
+        request.hostname = uri.host().map(|s| s.to_owned());
```

Comment:
Does this also get the case of when only one is set? Could you add a test to ensure this?